<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1654</width>
    <height>1031</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="main_tab">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1621</width>
      <height>971</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="Images">
     <attribute name="title">
      <string>Images</string>
     </attribute>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>20</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>slopes image</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>740</x>
        <y>30</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>phase screen</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>1230</x>
        <y>30</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>normalized psf</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>720</x>
        <y>470</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>phase residuals</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>480</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>dm phase</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_slopes_image_checkbox">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>450</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_phase_screen_checkbox">
      <property name="geometry">
       <rect>
        <x>730</x>
        <y>440</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_normalized_psf_checkbox">
      <property name="geometry">
       <rect>
        <x>1250</x>
        <y>460</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_phase_residuals_checkbox">
      <property name="geometry">
       <rect>
        <x>720</x>
        <y>890</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_dm_phase_checkbox">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>890</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QWidget" name="normalized_psf_widget" native="true">
      <property name="geometry">
       <rect>
        <x>1079</x>
        <y>79</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="phase_screen_widget" native="true">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>70</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="slopes_image_widget" native="true">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>70</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="dm_phase_widget" native="true">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>530</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="phase_residuals_widget" native="true">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>530</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>40</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>50</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>1240</x>
        <y>50</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>740</x>
        <y>500</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>510</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>1210</x>
        <y>530</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>scale λ</string>
      </property>
     </widget>
     <widget class="QWidget" name="dm_command_widget" native="true">
      <property name="geometry">
       <rect>
        <x>1060</x>
        <y>550</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_dm_command_checkbox">
      <property name="geometry">
       <rect>
        <x>1200</x>
        <y>900</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_23">
      <property name="geometry">
       <rect>
        <x>1200</x>
        <y>500</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>dm command</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="cam">
     <attribute name="title">
      <string>Cam</string>
     </attribute>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>30</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>cam 1</string>
      </property>
     </widget>
     <widget class="QWidget" name="cam1_widget" native="true">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_cam1_checkbox">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>460</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_24">
      <property name="geometry">
       <rect>
        <x>720</x>
        <y>30</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>cam 2</string>
      </property>
     </widget>
     <widget class="QWidget" name="cam2_widget" native="true">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>80</y>
        <width>471</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QCheckBox" name="autoscale_cam2_checkbox">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>460</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>autoscale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Vector">
     <attribute name="title">
      <string>Vector</string>
     </attribute>
     <widget class="QWidget" name="computed_act_pos_widget" native="true">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>40</y>
        <width>601</width>
        <height>371</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="computed_KL_modes_widget" native="true">
      <property name="geometry">
       <rect>
        <x>740</x>
        <y>40</y>
        <width>601</width>
        <height>371</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="commands_widget" native="true">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>510</y>
        <width>601</width>
        <height>361</height>
       </rect>
      </property>
     </widget>
     <widget class="QCheckBox" name="square_computed_act_pos_checkbox">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>430</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>square</string>
      </property>
     </widget>
     <widget class="QPushButton" name="reset_computed_act_pos_view_button">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>430</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset View</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="square_computed_KL_modes_checkbox">
      <property name="geometry">
       <rect>
        <x>1070</x>
        <y>440</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>square</string>
      </property>
     </widget>
     <widget class="QPushButton" name="reset_computed_KL_modes_view_button">
      <property name="geometry">
       <rect>
        <x>950</x>
        <y>440</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset View</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="square_commands_checkbox">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>880</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>square</string>
      </property>
     </widget>
     <widget class="QPushButton" name="reset_commands_view_button">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>880</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Reset View</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>20</y>
        <width>361</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Computed Actuator Positions Projected on KL Modes</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>980</x>
        <y>10</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Computed KL modes</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>460</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Commands</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="main">
     <attribute name="title">
      <string>Main</string>
     </attribute>
     <widget class="QDoubleSpinBox" name="gain_spinbox">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>10</y>
        <width>76</width>
        <height>27</height>
       </rect>
      </property>
      <property name="maximum">
       <double>2.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>990</x>
        <y>20</y>
        <width>41</width>
        <height>18</height>
       </rect>
      </property>
      <property name="text">
       <string>Gain</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>990</x>
        <y>50</y>
        <width>41</width>
        <height>18</height>
       </rect>
      </property>
      <property name="text">
       <string>Leak</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="leakage_spinbox">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>40</y>
        <width>76</width>
        <height>27</height>
       </rect>
      </property>
      <property name="maximum">
       <double>2.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
      <property name="value">
       <double>0.000000000000000</double>
      </property>
     </widget>
     <widget class="QSpinBox" name="delay_spinbox">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>80</y>
        <width>52</width>
        <height>27</height>
       </rect>
      </property>
      <property name="maximum">
       <number>194</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="num_iterations_spinbox">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>120</y>
        <width>91</width>
        <height>27</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>980</x>
        <y>80</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Delay</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>910</x>
        <y>120</y>
        <width>121</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Num iterations</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="calibration">
     <attribute name="title">
      <string>Calibration</string>
     </attribute>
     <widget class="QPushButton" name="start_calibration_button">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>80</y>
        <width>141</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Start Calibration</string>
      </property>
     </widget>
     <widget class="QWidget" name="calibration_gain" native="true">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>441</width>
        <height>321</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="linearity" native="true">
      <property name="geometry">
       <rect>
        <x>89</x>
        <y>459</y>
        <width>461</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="start_latency_identification_button">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>320</y>
        <width>231</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Start Latency identification</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>580</y>
        <width>52</width>
        <height>27</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="do_latency_estimation_2">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>540</y>
        <width>191</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Check mode linearity</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="latency_identification_output">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>370</y>
        <width>251</width>
        <height>131</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="save_calibration_button">
      <property name="geometry">
       <rect>
        <x>830</x>
        <y>80</y>
        <width>141</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Calibration</string>
      </property>
     </widget>
     <widget class="QPushButton" name="load_calibration_button">
      <property name="geometry">
       <rect>
        <x>990</x>
        <y>80</y>
        <width>141</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Load Calibration</string>
      </property>
     </widget>
     <widget class="QPushButton" name="save_latency_button">
      <property name="geometry">
       <rect>
        <x>940</x>
        <y>320</y>
        <width>131</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Latency</string>
      </property>
     </widget>
     <widget class="QPushButton" name="load_latency_button">
      <property name="geometry">
       <rect>
        <x>1100</x>
        <y>320</y>
        <width>191</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Load Latency</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="calibration_output">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>120</y>
        <width>251</width>
        <height>131</height>
       </rect>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="wait_time_spinbox">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>30</y>
        <width>76</width>
        <height>27</height>
       </rect>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>0.020000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>670</x>
        <y>30</y>
        <width>81</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>wait time</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Process</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1654</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
